src/clean-core/bit_cast.hh

```diff
@@ -10,6 +10,7 @@ template <class To, class From>
 {
     static_assert(std::is_trivially_copyable_v<To>, "only supported for trivially copyable types");
     static_assert(std::is_trivially_copyable_v<From>, "only supported for trivially copyable types");
+    static_assert(!std::is_same_v<To, From>, "identity bitcast");
```

Comment:
I wonder if this might be a bother in generic code where that can happen

Reply:
Agreed, i removed it again